src/pages/api/event/[id]/repairers/index.ts

```diff
@@ -0,0 +1,56 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import { z } from "zod";
+
+import apiHandler from "@/lib/api-handler";
+import prisma from "@/lib/prisma";
+import userService from "@/services/user";
+
+export default apiHandler({
+  get: getRepairers
+});
+
+async function getRepairers(req: NextApiRequest, res: NextApiResponse) {
+  const eventId = z.string().parse(req.query.id);
+  const result = await getEventRepairers(eventId as string);
+
+  res.status(200).json(result);
+}
+
+async function getEventRepairers(id: string) {
+  // get repairers from table EventRepairer
+  const repairers = await prisma.eventRepairer.findMany({
+    where: { eventId: id }
+  });
+
+  if (repairers.length === 0) {
+    return [];
+  }
+
+  const userIds = repairers.map((repairer) => repairer.userId);
+
+  // get additional informations like firstName, lastName, email
+  const repairersUsers = await getRepairersUsers(userIds);
+
+  // if user is not found in clerk, return just userId
+  return userIds.map((id) => {
+    const user = repairersUsers.find((user) => user.userId === id);
+
+    return {
+      userId: id,
+      firstName: user?.firstName ?? "",
+      lastName: user?.lastName ?? "",
+      email: user?.email ?? ""
+    };
+  });
+}
```

Comment:
Should we even be returning the user ids if the user is not found in clerk? Ideally, it should just ignore the unmatched user ids. 

Furthermore, there should not be a mismatch in clerk and the repairers table since we already have a check in the POST repairers .

Reply:
What we do here is : 
1. get event repaiere userIds from db;
2. get user data from clerk.

In the following scenarios, it could result in an invalid user:
1. userId in db is incorrect, like the data in table was altered;
2. clerk reurns invalid data, like the clerk service is down.

Filtering the userId if it is not found in the clerk could lead to the number of event repairers in the database not matching the number of event repairers returned from this API. This might be difficult for system maintainers to notice there are some issues in getting the user information from this service.
If we receive a response like {"user": userId, "firstName": "", "lastName": ""}, it becomes easy to recognize issues in retrieving this userId from the clerk, facilitating troubleshooting.

If we can ensure there are no mismatches between the clerk and the table in db, I will return the valid user data here and filter the invalid userIds.

Reply:
Those are some valid reasons. I just do not like how verbose and over-complicated the implementation looks currently.

To address some of your reasoning
- I think It is highly unlikely that there will be mismatches between clerk and the table in db, since people should not be allowed to be able to alter the database directly. 
- As for troubleshooting, should we maybe log it instead of returning the error to the user since it is for system maintainers to know and not the end user? 

A possible implementation
```javascript

const getRepairerUsers = (userIds: string[]) => {
  const users = await userService.getUsersByIds(userIds) // will probably to need to implement this or see if you could reuse something else
  
  if(users.length !== usersIds.length) {
    console.error('Mismatch in users from clerk and database');
  }
  
  return users;
}
```

Reply:
While logging is generally a reliable solution for service troubleshooting, as I mentioned in the associated issue earlier, I've implemented a check for user lengths in the latest commit. 

However, I am not sure whether using the console in the backend is a good practice, this might be something to consider later.